test/hotspot/jtreg/compiler/whitebox/TestEnqueueInitializerForCompilation.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8264395
+ * @summary testing of WB::enqueueInitializerForCompilation()
+ * @library /test/lib /
+ * @build sun.hotspot.WhiteBox
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller sun.hotspot.WhiteBox
+ * @run main/othervm/timeout=300 -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions
+ *      -XX:+WhiteBoxAPI compiler.whitebox.TestEnqueueInitializerForCompilation
+ */
+
+package compiler.whitebox;
+
+import sun.hotspot.WhiteBox;
+
+public class TestEnqueueInitializerForCompilation {
+
+    public static void main(String[] args) {
+        WhiteBox.getWhiteBox().enqueueInitializerForCompilation(LongWrapper.class, 4);
```

Comment:
What if compilation level 4 (= C2) is not available?

Reply:
Found unavailable/bad comp levels handled in WhiteBox::compile_method() or later in CompileBroker::compile_method.

Reply:
Right but wouldn't we hit an assert there if C2 is not available?

Reply:
For the case scenario mentioned found that the test simply displays a WB whitebox error that is ignored.
So seems we can go ahead with the current test case itself. Thanks

Reply:
Yes, as we've discussed off-thread, that's fine. Thanks for investigating.